public/component_types/indices/knn_index.ts

```diff
@@ -73,6 +75,7 @@ export class KnnIndex extends BaseComponent implements IComponent {
       // simple form inputs vs. complex JSON editor
```

Comment:
Have you seen a way to dynamically create a form based on a list of potential fields? Simple way would be key values that you can just keep adding for mapping but maybe something more complicated for nested fields?

Reply:
That's what this PR achieves. See the helper fns in [resizable_workspace](https://github.com/ohltyler/opensearch-ai-flow-dashboards-1/blob/form/public/pages/workflow_detail/workspace/resizable_workspace.tsx#L87-L124). Lmk if you're referring to something else.

Reply:
I assume for most cases the keys will be defined and I see how we read those from the fields defined for the components. I meant also for cases where we might have users add additional parameters that aren't defined or is that not the plan right now?